File: backend/src/main/java/com/h2o/h2oServer/domain/quotation/dto/QuotationRequestDto.java
package com.h2o.h2oServer.domain.quotation.dto;

import com.h2o.h2oServer.domain.model_type.dto.ModelTypeIdDto;
import lombok.Builder;
import io.swagger.annotations.ApiModel;
import lombok.Data;

import java.util.List;

@ApiModel(value = "견적 저장 요청")
@Data
@Builder
public class QuotationRequestDto {
    private Long carId;
    private Long trimId;
    private ModelTypeIdDto modelTypeIds;
    private Long internalColorId;
    private Long externalColorId;
    private List<Long> optionIds;
    private List<Long> packageIds;
}

File: iOS/Catalog/Catalog/Quotation/Data/DTO/QuotationRequestDTO.swift
//
//  QuotationRequestDTO.swift
//  Catalog
//
//  Created by 이수민 on 2023/08/21.
//

import Foundation

struct ModelTypeIDRequestDTO: Codable {
  var powertrainId: Int
  var bodyTypeId: Int
  var drivetrainId: Int
}

struct QuotationRequestDTO: Encodable {
  var carId: Int
  var modelTypeIds: ModelTypeIDRequestDTO
  var internalColorId: Int
  var externalColorId: Int
  var optionIds: [Int]
  var packageIds: [Int]
  var trimId: Int
}

